### DISCUSSION TITLE
v3.8.x Goals

### DISCUSSION THREAD

**Discussion by mokun** [mars-sim/mars-sim#1217] [General]
Currently, we have 38 open issues for v3.8.0. See https://github.com/mars-sim/mars-sim/milestones/3.8.0?page=2&q=is%3Aopen+milestone%3A3.8.0

Let's go rack and stack them first.

**Comment by bevans2000** [comment]
Sure let's focus on the tagged issues first

**Comment by bevans2000** [comment]
We need to improve the SonarQube results; recently the figures have gone significantly worse. I looked last night and there is an increase in commented out. Let's just remove it.
Set the goal of zero bugs.

**Reply by mokun** [reply]
So which one are you interested in working on ?

**Reply by mokun** [reply]
For me, my priority is fixing issues with construction missions so that a settlement may expand by having more more in-situ based buildings.

I'd like to assume that those brick-like buildings from the mining and trading outpost templates be treated as the first generation Mars-made buildings.

After a while, when the level of manufacturing capability is reasonably caught up, we may allow settlers to build hab-like buildings that are being sent to Mars.

We can even assume Earth's technological level of maturity in near terms will be always one step ahead of Mars. These advanced buildings may be  made on the Moon and be shipped to Mars.

**Reply by bevans2000** [reply]
That makes sense as Construction is important.
I will just work in the issues assigned to the milestone, probably #973 first.
Going to clean up the code in the console module as well

**Comment by mokun** [comment]
@bevans2000 

Currently, we have 25 open issues and 24 closed issues under [3.8.0](See https://github.com/mars-sim/mars-sim/milestone/15).

![image](https://github.com/mars-sim/mars-sim/assets/1168584/bc97e19f-98c2-4fcb-ab33-f9179aaef022)

Which ones would you like to finish and which ones to push to later release ?

**Reply by mokun** [reply]
@Urwumpe 

Would you like to name what feature you most want ? 

Which one or two issues would you like to get resolved ASAP ?

**Comment by bevans2000** [comment]
I will be closing #1285 today via a PR. I'm working on #274 but it's a bit of a bigger problem.
#1094 must be done by now?
I think 3.8.0 is full enough now and would just push the other to 3.9.0 BUT the latter would need a re-planning and some bigger ones going to 4.0

**Comment by bevans2000** [comment]
I think we need to close 3.8.0 soon. I would like to close #1315. 
In 3.9 we focus on the new Mission pattern?

**Reply by mokun** [reply]
Now i'm working on avoiding those premature cancelation of missions. There are a few associated problems such as getting stuck at departing and not reporting why it was unable to enter and leave garage, etc.

**Reply by mokun** [reply]
What are the main features of 3.8.0 ?

For now, I see we should at least : 

1. Fix the robot charging
2. Lessen the cancellation of missions
3. Fix the settlement/sponsor combo box issue in Monitor Tool
4. Fix the jumpiness of Settlement Map 
6. Avoid the amount of Methane going down to zero

**Reply by bevans2000** [reply]
So these are things to complete before closing the current 3.8 development?
Are you thinking there should be a release every 6 months? It's been 5 months since 3.7

**Reply by bevans2000** [reply]
Actually it's been 6 months since 3.7.0.
Move the issues out of the 3.8.0 milestone so we have those left you really want in 3.8 and we can focus on them

**Reply by mokun** [reply]
I just moved all remaining open issues to 3.9.0 and 4.0.0.

Now we have 26 issues closed in 3.8.0. See https://github.com/mars-sim/mars-sim/milestone/15.

**Reply by mokun** [reply]
We're now done to the remaining 3 tasks before releasing 3.8.0:

1. Fix the robot charging
2. Fix the settlement/sponsor combo box issue in Monitor Tool
3. Fix the jumpiness of Settlement Map

**Reply by bevans2000** [reply]
So there are no issue left in 3.8.0 so does that mean it is ready to go?

**Reply by mokun** [reply]
> no issue left in 3.8.0 so does that mean it is ready to go?

In my opinion, it's not ready yet. The following are to be fixed : 

1. Robot charging
2. Settlement/sponsor combo box issue in Monitor Tool
3. Jumpiness of Settlement Map
4. Avoid reaching zero methane

**Reply by bevans2000** [reply]
I have fixed the sorting in the Combo.
I don't understand the jumpiness or to robot charging problem. Why not create 2 new issues to cover them under 3.8.0

**Reply by mokun** [reply]
> the jumpiness

Say, if you are viewing the Settlement Map of settlement Foundation, in the first minute, everything works fine. However, after a minute or so, all of a sudden, without warning, the map will jump to settlement Arcadia (Arcadia is the 1st settlement in the combo box). 

So far, it only jumps to Arcadia by itself.

**Reply by mokun** [reply]
> robot charging problem

Robots still get depleted at 0% battery. It's not supposed to.

**Reply by bevans2000** [reply]
> > the jumpiness
> 
> Say, if you are viewing the Settlement Map of settlement Foundation, in the first minute, everything works fine. However, after a minute or so, all of a sudden, without warning, the map will jump to settlement Arcadia (Arcadia is the 1st settlement in the combo box).
> 
> So far, it only jumps to Arcadia by itself.

@mokun  I've found it and will commit a fix soon. The problem is the arrival of a new Settlement

**Reply by mokun** [reply]
>> So far, it only jumps to Arcadia by itself.
>The problem is the arrival of a new Settlement

I made a change to avoid calling setSettlement when an associated person is added during resupply mission. 
See https://github.com/mars-sim/mars-sim/commit/d1a450f39ebf95e3a8c58564e9196164cb186f32.

But it doesn't fix the jumpiness yet.

**Reply by bevans2000** [reply]
I found the problem and it should be fixed

**Reply by mokun** [reply]
What's the new process of releasing the binary ? Is it automated by GitHub Action yet ?

**Reply by bevans2000** [reply]
Just create a tag on the master branch and a draft release will be created containing the release artifacts.
Then add any notes and make it available
I need to update the wiki page.

**Reply by bevans2000** [reply]
Release number in poms need updating

**Reply by bevans2000** [reply]
Actually I have updated the release page in the wiki. I will update it this weekend

**Reply by bevans2000** [reply]
Wiki page updated https://github.com/mars-sim/mars-sim/wiki/Release-Procedure

**Reply by mokun** [reply]
In the past, I need to first generate a new set of HTML files and manually copy them from the target folder to the src folder. I don't think this step is automatic yet, right ?

Then I'll need to compile the binaries on my own machine and manually upload it to GitHub's new release page. Is this step already done away with ?

**Reply by bevans2000** [reply]
No that's all in the past. Look at the wiki page for release and that explains it. Basically create a tag and then push it. Then a specific Action will fire to built it and create a draft Release.

**Comment by bevans2000** [comment]
@mokun A few of the new recent UnitTests are randomly failing when run from GitHub. Locally they are running fine.
I'm trying to work through the problems but it's hard to understand why they fail

**Reply by bevans2000** [reply]
I think I've improved the situation with c95afe79a9221f34b2f307a992b005b5b4e08c12
Committed to master

**Reply by mokun** [reply]
It looks like some BuildingPanel* (such as BuildingPanelPower and BuildingPanelThermal) suffers from the shortening of the title description on the top : 

![image](https://github.com/mars-sim/mars-sim/assets/1168584/b7f00b1e-1709-4c1d-aff6-9ac8ac00bb5e)

![image](https://github.com/mars-sim/mars-sim/assets/1168584/1677d23d-f4f3-495c-ae93-ff152d504393)

On the other hand, BuildingPanelResourceProcessing has a longest title. Yet it's not affects by it.

![image](https://github.com/mars-sim/mars-sim/assets/1168584/e676a1f3-40b5-49ef-bd77-8bab973b96c9)


Why ?

**Reply by bevans2000** [reply]
That's odd

**Reply by mokun** [reply]
I just found out another scenario in which the title text, for instance, "Power Grid", would turn into its short form "Power Gr...". 

When I drag the whole main window from my external monitor back to the built-in LCD of my laptop., it automatically turn into its short terms. It's as if Flat LAF calculate the size of the text to try to fit in. 

It sounds it's over-doing it as if it's like a bug.

**Comment by mokun** [comment]
Are you still working on https://github.com/mars-sim/mars-sim/pull/1329 ? 

Finish it before releasing 3.8.0 ?

Turns out we did not work on Construction at all for 3.8.0.

**Reply by bevans2000** [reply]
Yes, I will commit something at weekend

**Reply by bevans2000** [reply]
What's left to do then in 3.8.0?

**Reply by Urwumpe** [reply]
How about maintaining a task board with the official goals and needed tasks for calling a release finished?

**Reply by bevans2000** [reply]
> How about maintaining a task board with the official goals and needed tasks for calling a release finished?

We have the Milestone in Issues and that should be the driving force. I believe it is now empty for 3.8.0, hence my question.
Anything we do should be tracked back to an issue

**Comment by mokun** [comment]
There's still a resource imbalance problem.

The methane will run out in about 20 sols for some settlements.

I've tried a few ways to avoid that but it still happens.

Since most buildings has to burn methane for heating , I'm going to investigate the rate and how much it's used for that.

Any insights ?

**Reply by mokun** [reply]
By the way, I've tried to tweak the methane demand value to go above the hydrogen demand value so as to ensure the 2 resource processes (Sabatier and the other) that produce methane as output would turn on.

Also, I've tried to avoid having the 2 other processes (PPA and the other) turn on to use up methane as input.

It's better now but still happens from time to time.

**Reply by Urwumpe** [reply]
Did you check of the maximum local production is at least close to the maximum local consumption?

**Reply by mokun** [reply]
> Did you check of the maximum local production is at least close to the maximum local consumption?

Yes to a certain degree. 

The stored amount of methane used to be dwindling down faster then being produced in the past. 

Recently, we've corrected that but it can still go to zero in long run.

On top of that, we haven't been able to track the exact amount of resource being consumed vs being produced to see they go in real-time yet. We have wanted to do that in https://github.com/mars-sim/mars-sim/issues/973.

If we can track where it goes exactly, we can debug further.

**Reply by bevans2000** [reply]
So you want to include #973 in v3.8.0? I know how to do this as have experienced with creating a specialised ResourceHolder that records the store and retrieve requests.
If you want this then I know how to do it.

**Reply by mokun** [reply]
> If you want this then I know how to do it.

Sure. Why not ? May as well include that :)

Meanwhile, I found big issues with the ThermalSystem/ThermalGeneration code.

So i'm in the process of reworking it to be able to make use of the fine-grained percentage of heat.

I'll get rid of the step-wise heat mode which cause too sudden jump or fall in temperature.

**Reply by bevans2000** [reply]
Need those unit tests to make sure it doesn't break and can test outside of the simulation

**Reply by mokun** [reply]
I've finished a bulk of work patching over the power and the thermal system. 

See below in that I beef up the use of the Building tab in Monitor Tool to show the stats of the thermal generation for each building.

![image](https://github.com/mars-sim/mars-sim/assets/1168584/8388d0d9-84ee-4ce3-9ee4-3a6316ab2fe4)

**Reply by bevans2000** [reply]
It will take me a while to do #973 so maybe we release 3.8.0 now

**Reply by mokun** [reply]
In terms of mission balance, it seems there's a crazy # of regolith prospecting missions starting. 

However, there are very few or no field missions. 

There are some exploration missions. But not enough. 

Rarely any mining missions taking place. Btw, Can you check if mining missions still have bugs ?

**Reply by mokun** [reply]
In v3.9.0, I'd like to pave the way for road constructions and building transportation hubs such as train stations, etc., in addition to weather stations and communication towers.

**Reply by bevans2000** [reply]
I suggest you create a Project with a set of steps to get what you want. Must use some branching if these are big changes.
I want to focus on the concurrency issue when Trade and Delivery missions, we have an issue.
Then work on the generic Mission approach

**Reply by bevans2000** [reply]
I also have an idea about changing how we capture metrics and centralise that.

**Reply by bevans2000** [reply]
> In terms of mission balance, it seems there's a crazy # of regolith prospecting missions starting. 
> 
> However, there are very few or no field missions. 
> 
> There are some exploration missions. But not enough. 
> 
> Rarely any mining missions taking place. Btw, Can you check if mining missions still have bugs ?
> 

I think we focus on the new project based approach unless it breaks

**Comment by mokun** [comment]
I try running HelpGenerator but it throws me error : 

```
C:\Users\mkhelios\eclipse\java-2024-06\git\mars-sim\mars-sim\mars-sim-tools\src\main\java\com\mars_sim\tools\helpgenerator>java com.mars_sim.tools.helpgenerator.HelpGenerator ../mars-sim-ui/src/main/resources/docs/help/generated    
Error: Could not find or load main class com.mars_sim.tools.helpgenerator.HelpGenerator
Caused by: java.lang.ClassNotFoundException: com.mars_sim.tools.helpgenerator.HelpGenerator
```

**Reply by mokun** [reply]
Can you see if you can run it to produce the html pages ? 

Also, is the destination supposed to be inside the folder called `generated` ? 

I suppose that folder will be created, right ? 

I have made some changes to manufacturing.xml and resources.xml, etc., 

Q: do I need to run it manually in order for it to produce a new set of htmls prior to each release ?

**Reply by mokun** [reply]
Instead of mars-sim-ui, may be you mean to say 

`/mars-sim-tools/src/main/resources/help/generated` ?

**Reply by bevans2000** [reply]
> I try running HelpGenerator but it throws me error : 
> 
> ```
> C:\Users\mkhelios\eclipse\java-2024-06\git\mars-sim\mars-sim\mars-sim-tools\src\main\java\com\mars_sim\tools\helpgenerator>java com.mars_sim.tools.helpgenerator.HelpGenerator ../mars-sim-ui/src/main/resources/docs/help/generated    
> Error: Could not find or load main class com.mars_sim.tools.helpgenerator.HelpGenerator
> Caused by: java.lang.ClassNotFoundException: com.mars_sim.tools.helpgenerator.HelpGenerator
> ```
> 
> 

You can't run it like that. It is like any other of our main classes and needs to other classes and libraries. There is a launcher file I think

**Reply by bevans2000** [reply]
> Instead of mars-sim-ui, may be you mean to say 
> 
> `/mars-sim-tools/src/main/resources/help/generated` ?

No because we do.not commit the generated file to got any more.

**Reply by bevans2000** [reply]
> Can you see if you can run it to produce the html pages ? 
> 
> Also, is the destination supposed to be inside the folder called `generated` ? 
> 
> I suppose that folder will be created, right ? 
> 
> I have made some changes to manufacturing.xml and resources.xml, etc., 
> 
> Q: do I need to run it manually in order for it to produce a new set of htmls prior to each release ? 
> 

I documented the new approach on the commit. The html is generated on demand when the help tool is used from the UI. It auto generated based on the release number.
But if you want to force a regenerate then delete the docs folder inside the local.mars SIM folder

**Reply by mokun** [reply]
> You can't run it like that. It is like any other of our main classes and needs to other classes and libraries. There is a launcher file I think

There's that `generate-help.md` file you put inside mars-sim-tools. It tells me to run it that way.

```
To generate the UI help use the Help Generator tool contained within. It takes a single argument that is the target directory.

> java com.mars_sim.tools.helpgenerator.HelpGenerator ../mars-sim-ui/src/main/resources/docs/help/generated
```

**Comment by mokun** [comment]
> The html is generated on demand when the help tool is used from the UI. It auto generated based on the release number.

So when you say "generated on demand", do you mean that during the simulation, if I click on a process, it will give me the specs for that process. 

Behind the scene, a temporary html file is generated and be read and converted into a tooltip in Swing UI ?

**Reply by mokun** [reply]
I'm re-reading https://github.com/mars-sim/mars-sim/issues/1260 now to understand what's been changed.

**Comment by mokun** [comment]
We can release 3.8.0 now if there's nothing else critical to fix.

**Reply by mokun** [reply]
argh I just found another critical bug in Heating in calculating required heat. On the way fixing it now.

**Reply by mokun** [reply]
> bug in Heating in calculating required heat.

This one is now fixed.

However, some vehicles have trouble finishing their maintenance tasks.

![image](https://github.com/mars-sim/mars-sim/assets/1168584/11b5e440-ae48-4584-a088-cb2a48a61fc1)

They never got done. SVG pics shown parts were outside vehicles.

Also, two vehicles occupied in a garage. They don't look like under maintenance.

If they're not, they should be left and make room for the other vehicles that are in need of maintenance.

Somehow because of all these, there has been zero missions for over 30 days.

![image](https://github.com/mars-sim/mars-sim/assets/1168584/43df4d64-66cb-442b-9d35-db2e26808fd0)

Also, the two vehicles that went out on Ice prospecting never came home.

It also fails to start Emergency missions because vehicles are not available somehow.

```
01-Bora-06:224.299 (Info) MissionManager : [Hartmann Base - Command and Control 1] Olivia Karlsson - Emergency Supply Score: 17.25 (base: 20, Role: 1, settlementratio: 1, leader: 0.862)
01-Bora-06:224.299 (Info) AbstractMission : [Hartmann Base - Command and Control 1] Olivia Karlsson - Began organizing an Emergency Supply 34-004 mission.
01-Bora-06:224.299 (Info) AbstractMission : [Hartmann Base - Command and Control 1] Olivia Karlsson - Ended the Emergency Supply 34-004 with the status flag(s): No available vehicles.
01-Bora-06:224.299 (Info) AbstractMission : [Hartmann Base - Command and Control 1] Olivia Karlsson - Disbanding mission member(s): Olivia Karlsson
01-Bora-06:224.299 (Warning) AbstractVehicleMission : [Hartmann Base - Command and Control 1] Olivia Karlsson - Could not reserve a vehicle for Emergency Supply 34-004.
```

**Reply by mokun** [reply]
2 vehicles in garage also could not finish unloading CO2 in the last 30 sols:
![image](https://github.com/mars-sim/mars-sim/assets/1168584/cb045968-1853-4716-a51d-d39ba1af4204)
![image](https://github.com/mars-sim/mars-sim/assets/1168584/7faee74f-88a0-44a9-a1c2-f730bbd20f18)

3 outside vehicles are very faraway and supposedly under maintenance:
![image](https://github.com/mars-sim/mars-sim/assets/1168584/52927ddd-917c-411d-b267-25ed337289d9)
![image](https://github.com/mars-sim/mars-sim/assets/1168584/d1b292b6-72b9-4914-b3f5-9259c851cabd)

**Reply by mokun** [reply]
Some walking issues : 

```
01-Adir-05:043.718 (Severe) WalkingSteps [x53] : [Hartmann Base - Wind Runner] Afonso Jesus - Invalid destination (-150.46, 69.92) not within rover Wind Runner @ (-3.3, 21.2)(3.5x9.5) @ 125.49665927927818
01-Adir-05:043.718 (Severe) WalkingSteps [x53] : [Hartmann Base - Wind Runner] Gun-woo Chew - Invalid destination (-92.63, -31.19) not within rover Wind Runner @ (-3.3, 21.2)(3.5x9.5) @ 125.49665927927818
01-Adir-05:043.718 (Severe) WalkingSteps [x53] : [Hartmann Base - Wind Runner] Avika Aggarwal - Invalid destination (-92.41, -31.33) not within rover Wind Runner @ (-3.3, 21.2)(3.5x9.5) @ 125.49665927927818
01-Adir-05:043.718 (Severe) WalkingSteps [x53] : [Hartmann Base - Wind Runner] Maria Kowalski - Invalid destination (-9.23, -0.47) not within rover Wind Runner @ (-3.3, 21.2)(3.5x9.5) @ 125.49665927927818
01-Adir-05:043.718 (Severe) WalkingSteps [x53] : [Hartmann Base - Wind Runner] He-Huang Zhou - Invalid destination (-91.79, -31.35) not within rover Wind Runner @ (-3.3, 21.2)(3.5x9.5) @ 125.49665927927818
01-Adir-05:043.718 (Severe) WalkingSteps [x53] : [Hartmann Base - Wind Runner] Máté Molnár - Invalid destination (-92.48, -31.56) not within rover Wind Runner @ (-3.3, 21.2)(3.5x9.5) @ 125.49665927927818

...
01-Adir-05:052.591 (Severe) WalkingSteps [x4] : [Hartmann Base - Desert Fox] Paulo carvalho - Invalid destination (-91.33, -31.26) not within rover Desert Fox @ (85.5, -180.8)(3.5x8.0) @ 25.295363616360238
01-Adir-05:052.591 (Severe) WalkingSteps [x4] : [Hartmann Base - Desert Fox] Minato Yamada - Invalid destination (-149.54, 68.17) not within rover Desert Fox @ (85.5, -180.8)(3.5x8.0) @ 25.295363616360238
01-Adir-05:052.591 (Severe) WalkingSteps [x4] : [Hartmann Base - Desert Fox] Celia Ramos - Invalid destination (-93.83, -31.47) not within rover Desert Fox @ (85.5, -180.8)(3.5x8.0) @ 25.295363616360238
01-Adir-05:052.591 (Severe) WalkingSteps [x4] : [Hartmann Base - Desert Fox] Marcos Moraes - Invalid destination (-150.13, 70.19) not within rover Desert Fox @ (85.5, -180.8)(3.5x8.0) @ 25.295363616360238

```

**Reply by mokun** [reply]
I've fixed those parking issues at https://github.com/mars-sim/mars-sim/commit/372a5c2aaf608efad15df8caabc2d6c0eb09450f.

Now we must resolve this [new issue](https://github.com/mars-sim/mars-sim/issues/1347).

Can you at least take a look ?

**Reply by bevans2000** [reply]
Are we ready to close this off now?

**Reply by mokun** [reply]
3 last issues : 
1. Drone's navigation not working. 
2. Mining not working.
3. Troubleshoot why teleportation occurs.  
4. Wrong amount of collected resources.

**Reply by bevans2000** [reply]
Might be worth waiting for the new Mission approach to number 2

**Reply by bevans2000** [reply]
Can we close the remaining 3.8 issues then?

**Reply by mokun** [reply]
> Can we close the remaining 3.8 issues then?

Finding clues on cracking 1-3.

btw, I just add the 4th one above and I now know why the collected amount is always wrong. Fixing it.

**Reply by bevans2000** [reply]
I am going to be away on holiday for a few weeks so will leave it to you to close down 3.8

**Reply by mokun** [reply]
oh ok. When releasing 3.8.0 under the new procedures, 

Do I just need to 
1. Go to https://github.com/mars-sim/mars-sim/releases
2. Click on `Draft a new release`
3. Type in v3.8.0 and copy and paste content text and hit Enter.

then will GitHub Action do the rest for us ? 

No need of compiling the binary ourselves and uploading them ? 

No need of manually generating any html files ourselves ?

**Reply by bevans2000** [reply]
No it's simpler. You just need to create a tag. It's documented in the Wiki https://github.com/mars-sim/mars-sim/wiki/Release-Procedure

**Reply by mokun** [reply]
Since I don't use the command line to do git, in the next few days, I'll just manually create a release by drafting a tag with v3.8.0 and publish it. I suppose it'll have the same effect, right ?

**Reply by bevans2000** [reply]
You should be able to create a tag via Eclipse as well

**Reply by bevans2000** [reply]
When will 3.8.0 be closed off?

**Reply by mokun** [reply]
> When will 3.8.0 be closed off?

Say after we finish these 2 : 
1. https://github.com/mars-sim/mars-sim/issues/1371
2. https://github.com/mars-sim/mars-sim/issues/1372

It doesn't seem we can resolve this:
1. https://github.com/mars-sim/mars-sim/issues/1356

**Reply by bevans2000** [reply]
#1371 yes I totally agree. I will look at this next
#1372 I'm not sure about this because it fundamentally changes how we select Task but let's discuss in the issue

#1356 I have an idea how to trap this when it happens.

**Reply by mokun** [reply]
> https://github.com/mars-sim/mars-sim/issues/1356 I have an idea how to trap this when it happens.

May I ask in what ways ?

**Reply by bevans2000** [reply]
> > https://github.com/mars-sim/mars-sim/issues/1356 I have an idea how to trap this when it happens.
> 
> May I ask in what ways ?

Check in the transfer method that the vehicle is in a Settlement when a Person is transferred to a Settlement

**Comment by bevans2000** [comment]
Only #1372 is left but we should leave that as it's a simple  change so let's leave it out. So can we close 3.8 down now?
I feel we are just messing around now.

**Reply by mokun** [reply]
>  So can we close 3.8 down now?

Yes. I just updated the release date on htmls. 

Can you now go ahead to do a push for the 3.8.0 release ?

Thanks !

**Reply by mokun** [reply]
> https://github.com/mars-sim/mars-sim/issues/1356 I have an idea how to trap this when it happens.

Do you still wanna resolve #1356 before releasing 3.8.0 ?

**Reply by bevans2000** [reply]
I think it's part of the bigger change of better control of containment. That needs a clear approach not a number of small fixes.
So no I feel we should leave it out.
We need a clear discussion on what to focus on for 3.9 and stick to it as a target.

**Reply by mokun** [reply]
> We need a clear discussion on what to focus on for 3.9 and stick to it as a target.

It'll be nice to have a singular focus and put forth all the effort in adding, say, 1 major feature and 2 or 3 minor features in each new releases.

But in reality, we tend to deviate from it because we need to wait out until we have done some refactoring and address some imminent concerns in order to proceed to that major feature.

I'll my best to adhere to the plan !

**Reply by mokun** [reply]
I just did a commit to comment out excessive Heating-related logs. 

Anything else before the official release of 3.8.0 ?

Do you wanna go ahead to do a push since you're handy with that ?

**Reply by mokun** [reply]
Done ! 

I've released v3.8.0 the old fashion way. 

I wasn't using git directly to trigger a release since I don't have the git command line configured and not sure about how to use eclipse IDE to push differently.  Will try it next time.

**Reply by mokun** [reply]
> Done !
> 
> I've released v3.8.0 the old fashion way.
> 
> I wasn't using git directly to trigger a release since I don't have the git command line configured and not sure about how to use eclipse IDE to push differently. Will try it next time.

One thing I just notice is that v3.8.0 Swing Jar is 75 MB (vs. v3.7.2's 59MB).

Not sure why it's that big.

Do you want to pull the v3.8.0 back and skim off whatever made it so bloated ?

**Reply by mokun** [reply]
Also, I just discovered that a GitHub Action was somehow triggered, which leads to creating another entry called [Release v3.8.0](https://github.com/mars-sim/mars-sim/releases/tag/untagged-4de4465e6052914f9a3d). It's still in Draft form, awaiting for finalizing it. 

However, when I look inside that BIG ZIP file (127 MB), it has 3 folders as follows : 
![image](https://github.com/user-attachments/assets/c5ff5d7a-5fbe-48e0-bd40-e5361c10fa53)

One problem is that it has both the mars-sim-console.jar and the mars-sim-swing.jar inside.
![image](https://github.com/user-attachments/assets/34704977-58d5-4982-97c7-bf9cd263e5bd)

Should we give the player the choice to download the jar file they want ? instead of having both jarfiles in one big download ?

**Reply by mokun** [reply]
Also, there's a README.txt file in the root folder. 

Do we want to rename it to something else ? 

We already have a file called README.md (in the mars-sim folder) that GitHub would use as the first page when loading https://github.com/mars-sim/mars-sim/, right ?

**Reply by mokun** [reply]
I just created a [3.8.x](https://github.com/mars-sim/mars-sim/tree/3.8.x) branch for potential bugfixes.

**Reply by bevans2000** [reply]
> > Done !
> > I've released v3.8.0 the old fashion way.
> > I wasn't using git directly to trigger a release since I don't have the git command line configured and not sure about how to use eclipse IDE to push differently. Will try it next time.
> 
> One thing I just notice is that v3.8.0 Swing Jar is 75 MB (vs. v3.7.2's 59MB).
> 
> Not sure why it's that big.
> 
> Do you want to pull the v3.8.0 back and skim off whatever made it so bloated ?

The large JAR is because you added more map images files to the resources/map folder and these get bundled into the JAR. We added the download maps feature to solve this problems; only the minimum map were added to the JAR.

**Reply by bevans2000** [reply]
> Also, there's a README.txt file in the root folder.
> 
> Do we want to rename it to something else ?
> 
> We already have a file called README.md (in the mars-sim folder) that GitHub would use as the first page when loading https://github.com/mars-sim/mars-sim/, right ?

This is the README.txt that will be installed as part of the deployment; it's  not used by GitHub.

**Reply by bevans2000** [reply]
> Also, I just discovered that a GitHub Action was somehow triggered, which leads to creating another entry called [Release v3.8.0](https://github.com/mars-sim/mars-sim/releases/tag/untagged-4de4465e6052914f9a3d). It's still in Draft form, awaiting for finalizing it.
> 
Correct, the actions triggers on a tag of the correct format being created. The objective was to remove having to compile locally and then uploading the artefacts because this has the risk that a local change not commit may get added.
> 
> Should we give the player the choice to download the jar file they want ? instead of having both jarfiles in one big download ?
Yes we decided to have a single download ZIP which contains scripts for both Linux & Windows as a single download. It makes more sense and is easier.

**Comment by mokun** [comment]
It seems the resolution level switching issue in Mars Navigator is finally being identified and fixed.

What else would you like to include in 3.8.1 ?

**Reply by mokun** [reply]
But I don't know if I could make adjustment to the landmark text color in time but will try.

Would you like to try tackling on [backlog tasks](https://github.com/mars-sim/mars-sim/issues/1372) choice ?

**Reply by bevans2000** [reply]
Usually a point release is used for fixing bugs uncovered since a release was done. So I see no need to pick anything off the backlog. 
I'm working on completing #1324.

**Reply by mokun** [reply]
> Usually a point release is used for fixing bugs uncovered since a release was done.

Right. 

The loading/unloading issue was there all along. It's not trivial and requires design change, right ?

**Reply by bevans2000** [reply]
Sorry, #1324 is on the master branch for 3.9

**Reply by mokun** [reply]
@bevans2000 @Urwumpe @muybien3433 

Anything you see in 3.8.0 that needs to be fixed and include in 3.8.1 ?
